### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by manjari25**
This commit adds an unreleased section to the changelog 
to track things that need to be released. It also lists the
unreleased changes since the last release to this section.

### CODE REVIEW

**Review by abhinav — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
 # Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
```
Comment (abhinav) [inline CHANGELOG.md:5]: So this changelog doesn't quite adhere to keepachangelog.

The format here is a bit bespoke at this point. 

Unless we want to go through the trouble of migrating the format, I would only mention that we follow semver.

Comment (manjari25) [reply CHANGELOG.md]: Updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,15 @@
 # Changelog
+All notable changes to this project will be documented in this file.
+
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+### Fixed
+* [#1001][]: Fix complex number encoding with negative imaginary part. Thanks to @hemantjadon.
+* [#1003][]: Fix float32 encoding.
```
Comment (abhinav) [inline CHANGELOG.md:9]: Something more specific here? It's not that float32 was entirely broken; only that the preciseness was lost.

```suggestion
* [#1003][]: JSON: Fix inaccurate precision when encoding float32.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,15 @@
 # Changelog
+All notable changes to this project will be documented in this file.
+
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+### Fixed
+* [#1001][]: Fix complex number encoding with negative imaginary part. Thanks to @hemantjadon.
```
Comment (abhinav) [inline CHANGELOG.md:8]: Let's prefix this with component:

```suggestion
* [#1001][]: JSON: Fix complex number encoding with negative imaginary part. Thanks to @hemantjadon.
```

**Review by abhinav — APPROVED** [review]

**Pull request merged**